src/Features/CSharp/Portable/Completion/CompletionProviders/CompletionUtilities.cs

```diff
@@ -188,30 +188,53 @@ public static SyntaxNode GetTargetCaretPositionForMethod(BaseMethodDeclarationSy
         }
     }
 
-    public static SyntaxNode GetTargetCaretNodeForInsertedMember(SyntaxNode caretTarget)
+    public static TextSpan GetTargetSelectionSpanForMethod(BaseMethodDeclarationSyntax methodDeclaration)
+    {
+        if (methodDeclaration.ExpressionBody is not null)
+        {
+            // select the expression span
+            return methodDeclaration.ExpressionBody.Expression.Span;
+        }
+        else if (methodDeclaration.Body is not null)
+        {
+            // select the last statement in the method
+            return methodDeclaration.Body.Statements.Last().Span;
+        }
+        else
+        {
+            return methodDeclaration.Span;
```

Comment:
Is this path reachable?

Reply:
I don't think so, but I wasn't sure. I saw some places that throw under that assumption and others that handle that case. I decided to be safe and just return the method decl's span if this does happen.